### ISSUE TITLE
Mage registry key "_singleton/turpentine/observer_esi" already exists

### ISSUE THREAD

**Issue by Xavier027** [nexcess/magento-turpentine#1003]
Hello, I got this error after tryng to compile my eshop and having Magento-Turpentine + Varnish freshly installed. 
I can't find anyway to debug it, any help please?

Error report thrown:

``` 
There has been an error processing your request

Mage registry key "_singleton/turpentine/observer_esi" already exists

Trace:
#0 /var/www/vhosts/s479695541.onlinehome.fr/dragees-et-chocolats/app/Mage.php(223): Mage::throwException('Mage registry k...')
#1 /var/www/vhosts/s479695541.onlinehome.fr/dragees-et-chocolats/app/Mage.php(477): Mage::register('_singleton/turp...', false)
#2 /var/www/vhosts/s479695541.onlinehome.fr/dragees-et-chocolats/includes/src/__default.php(21667): Mage::getSingleton('turpentine/obse...')
#3 /var/www/vhosts/s479695541.onlinehome.fr/dragees-et-chocolats/app/Mage.php(448): Mage_Core_Model_App->dispatchEvent('controller_fron...', Array)
#4 /var/www/vhosts/s479695541.onlinehome.fr/dragees-et-chocolats/includes/src/__default.php(17889): Mage::dispatchEvent('controller_fron...', Array)
#5 /var/www/vhosts/s479695541.onlinehome.fr/dragees-et-chocolats/includes/src/__default.php(21100): Mage_Core_Controller_Varien_Front->init()
#6 /var/www/vhosts/s479695541.onlinehome.fr/dragees-et-chocolats/includes/src/__default.php(21445): Mage_Core_Model_App->_initFrontController()
#7 /var/www/vhosts/s479695541.onlinehome.fr/dragees-et-chocolats/includes/src/__default.php(20705): Mage_Core_Model_App->getFrontController()
#8 /var/www/vhosts/s479695541.onlinehome.fr/dragees-et-chocolats/app/Mage.php(684): Mage_Core_Model_App->run(Array)
#9 /var/www/vhosts/s479695541.onlinehome.fr/dragees-et-chocolats/index.php(99): Mage::run('', 'store')
#10 {main}
``` 
I'm getting this error from main page to any page in the frontend & backend. I can't login to the backend anymore

The Magento Compiler is disabled and cleared, the cache is emptied.

How can I solve this?

Thanks in advance

**Comment by aricwatson** [comment]
What version of Magento, Turpentine and Varnish are you using? Was Magento installed and working smoothly before you tried to set up Turpentine?

**Comment by Xavier027** [comment]
Hi Aricwatson,

I'm using the following Version of Varnish: varnishd (varnish-3.0.7 revision f544cd8)
Magento: ver. 1.9.0.1
Turpentine: ver. 0.6.1

I had Magento installed and working smoothly and had the Version of Turpentine installed since months.

1 week ago, I switched from Apache to Nginx & Varnish. Everything seemed to work smoothly, page load was faster too... until I tried to compile the shop.

I've seen that my version of Turpentine was old and tried to uninstall Turpentine afterwards with the downloader but I got the following error: "Connection string is empty".

Do you have an idea where the issue could be coming from?

**Comment by miguelbalparda** [comment]
Have you checked https://github.com/nexcess/magento-turpentine/search?utf8=%E2%9C%93&q=Mage+registry+key+%22_singleton%2Fturpentine%2Fobserver_esi%22+already+exists&type=Issues ?

**Comment by Xavier027** [comment]
Hi miguelbalparda,

Yes I did, I tried again everything this Weekend and I still have the same issue :-(. If somebody has an idea... :-(

https://github.com/nexcess/magento-turpentine/issues/682
I edited app/etc/modules/Nexcessnet_Turpentine.xml and changed:

``` 
    <modules>
        <Nexcessnet_Turpentine>
            <active>true</active>
            <codePool>community</codePool>
        </Nexcessnet_Turpentine>
    </modules>
</config>

``` 
to

``` 
<config>
    <modules>
        <Nexcessnet_Turpentine>
            <active>false</active>
            <codePool>community</codePool>
        </Nexcessnet_Turpentine>
    </modules>
</config>
``` 
And cleared the Cache. It doesn't work.

https://github.com/nexcess/magento-turpentine/issues/830
I cleared and disabled the Compiler
``` 
php -f shell/compiler.php clear
php -f shell/compiler.php disable
``` 

https://github.com/nexcess/magento-turpentine/issues/532
I removed the following file:
 ``` 
app/code/local/Mage/Core/Model/Session.php 
``` 
Then I tried to remove
``` 
app/code/community/Nexcessnet/Turpentine 
app/etc/modules/Nexcessnet_Turpentine.xml
``` 
And I cleared the Cache, without any effect.

https://github.com/nexcess/magento-turpentine/issues/144
The Magento Compiler is disabled and cleared, the cache is emptied and i'm still not able to access the config panel.

https://github.com/nexcess/magento-turpentine/issues/258
I'm not using MagentoGo :-(

**Comment by miguelbalparda** [comment]
Try disabling every cache but the 2 turpentine ones. Clear cache and restart your web services

**Comment by Xavier027** [comment]
Hi miguelbalparda, 
I disabled the following Cache:
``` 
app/etc/modules/Mage_PageCache.xml
``` 
I cleared the cache, restarted nginx & varnish but still have the same issue.

**Comment by miguelbalparda** [comment]
I was meaning disabling all the caches from System - Cache and leave only the Turpentine caches on. Please try that and refresh everything. Also, how are you using PHP?

**Comment by Xavier027** [comment]
Hi miguelbalparda,

it didn't work. :-(

Therefore I tried to figure out what could have been wrong in the compilation and checked my nginx and varnish configuration files.

I found out that the installation of Nginx & Varnish did remove the change the user with the write rights of magento. I think this is what created the bug when I tried to compile Magento.

I have done the following:

Put www-data as nginx user

``` 
user              www-data;
``` 

and put the rights to the magento folder:

``` 
chown -R www-data:psacln [FOLDER]
``` 

As the downloader is still available (well it's the only thing available), I tried again to uninstall Turpentine with the interface and it worked this time.

Once again I cleared and disabled the compiler and cleared the cache

``` 
rm -rf var/cache/*
``` 
Now the website is available :-). Works too once I have installed the new version of Turpentine.
